src/main/scala/wdl4s/types/WdlPairType.scala

```diff
@@ -19,6 +24,28 @@ case class WdlPairType(leftType: WdlType, rightType: WdlType) extends WdlType {
   override protected def coercion: PartialFunction[Any, WdlValue] = {
     case otherPair @ WdlPair(otherValue1, otherValue2) if isCoerceableFrom(otherPair.wdlType) =>
       WdlPair(leftType.coerceRawValue(otherValue1).get, rightType.coerceRawValue(otherValue2).get)
+    case jsObject: JsObject if jsObject.fields.size == 2 => mapToPair(jsObject.fields, this)
+  }
+
+  def mapToPair(m: Map[String, JsValue], wdlPairType: WdlPairType): WdlPair = {
+
+    val caseNormalizedMap = m map { case(k, v) => k.toLowerCase.capitalize -> v }
+
+    val wdlPair: Seq[Try[WdlValue]] = (caseNormalizedMap.get("Left"), caseNormalizedMap.get("Right")) match {
+      case (Some(leftVal), Some(rightVal)) => Seq(wdlPairType.leftType.coerceRawValue(leftVal), wdlPairType.rightType.coerceRawValue(rightVal))
+    }
+
+    val failures = wdlPair collect { case f:Failure[_] => f }
+
+    failures match {
+      case f: Iterable[Failure[_]] if f.nonEmpty =>
+        throw new UnsupportedOperationException(s"Failed to coerce one or more values for creating a ${wdlPairType.toWdlString}:\n${TryUtil.stringifyFailures(f)}}")
+      case _ =>
+        val coercedPair = wdlPair match { case Seq(left, right) => (left.get, right.get) }
```

Comment:
ToL: This `match` is also a bit convoluted. I think what you really want is just a one-liner, `WdlPair(wdlPair._1.get, wdlPair._2.get)`

EDIT: but this requires `wdlPair` to be a tuple. Which then ruins the `wdlPair.collect`. Sigh. If you want, there's a library that can flip tuples into a List, but we might be getting much further away from simple review changes at this point. From [this post](https://stackoverflow.com/questions/1919009/are-there-any-methods-included-in-scala-to-convert-tuples-to-lists):

> Using Shapeless -
```
@ import syntax.std.tuple._
import syntax.std.tuple._
@ (1,2,3).toList
res21: List[Int] = List(1, 2, 3)
@ (1,2,3,4,3,3,3,3,3,3,3).toList
res22: List[Int] = List(1, 2, 3, 4, 3, 3, 3, 3, 3, 3, 3)
```